### ISSUE TITLE
double quotes inside interpolated string parameter doesn't wokr

### ISSUE THREAD

**Issue by colemickens** [nushell/nushell#10956]
### Describe the bug

```
let x = [ "foo" "bar" ]
print -e $"($x | str join " ")"
```

does not work, I feel like it should.

command line parser treats it as invalid and doesn't let me send it. In a script, it complains that there should have been a closing paren.

### How to reproduce

See above.

### Expected behavior

see above.

### Configuration

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.86.1                                                            |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.73.0 (cc66ad468 2023-10-03) (built from a source tarball) |
| cargo_version      | cargo 1.73.0                                                      |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| allocator          | mimalloc                                                          |
| features           | default, sqlite, trash, which, zip                                |
| installed_plugins  |                                                                   |

**Comment by fdncred** [comment]
I don't believe it's ever worked. I remember writing in some documentation about quotes within quotes and how they have to be different. It is indeed a pain but I'm not sure how one could fix it. Nushell has to know where begin and end quotes are.

**Comment by colemickens** [comment]
Bash can do it... :pleading_face: Not that I'm volunteering to dive into the parser.

```
bash> echo "this is so cool $(echo "foo") trailer"
```

**Comment by fdncred** [comment]
we're not bash compatible 🤣 

I know that's not an answer, of course, please feel free to dive in and try to fix things a little bit to make them better. I agree that our quoting can be annoying sometimes. I especially hate backticks myself, but that's a different subject.

**Comment by amtoine** [comment]
i think this should be able to work.
because the inner `" "` is inside a subexpression, it looks like there is no ambiguity with the outside `$" ... "` :yum:
